src/bun.js/node/node_fs.zig

```diff
@@ -5215,8 +5229,14 @@ pub const NodeFS = struct {
 
             switch (comptime ExpectedType) {
                 Dirent => {
+                    const path_u8 = bun.path.dirname(bun.path.join(&[_]string{ root_basename, name_to_copy }, .auto), .auto);
+                    if (dirent_path_prev == null or bun.strings.eql(dirent_path_prev.?.byteSlice(), path_u8)) {
```

Comment:
@nektro was this line supposed to be?

```suggestion
                    if (dirent_path_prev == null or !bun.strings.eql(dirent_path_prev.?.byteSlice(), path_u8)) {
```